Server/ReasnAPI/ReasnAPI/Services/ImageService.cs

```diff
@@ -1,16 +1,145 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using static System.Net.Mime.MediaTypeNames;
+using Image = ReasnAPI.Models.Database.Image;
+
+namespace ReasnAPI.Services;
+public class ImageService(ReasnContext context)
+{
+    public List<ImageDto> CreateImages(List<ImageDto> imageDtos)
+    {
+        var newImages = new List<Image>();
+
+        foreach (var imageDto in imageDtos)
+        {
+            var image = context.Images.FirstOrDefault(r => r.ObjectId == imageDto.ObjectId && r.ObjectTypeId == imageDto.ObjectTypeId);
+            if (image != null)
+            {
+                continue;
+            }
+
+            var newImage = new Image
+            {
+                ImageData = imageDto.ImageData,
+                ObjectId = imageDto.ObjectId,
+                ObjectTypeId = imageDto.ObjectTypeId
+            };
+
+            newImages.Add(newImage);
+        }

```

Comment:
Can we also do it in AddRange? Select those images which are not already on DB and add them at once.